### PR TITLE
Improvement - Change HelpWindow

### LINKED ISSUE #215: Improvement - Change HelpWindow
To change the current HelpWindow to something more helpful

### PR DESCRIPTION
**Pull request by Wei-Xinn**
## Changes
This PR changes the HelpWindow to show command actions, formats and examples.
The commands are split into pages based on functionality.

image

image

## Closes

- closes #215

### CODE REVIEW

File: src/main/java/seedu/contax/ui/HelpWindow.java [hunk 1]
```diff
@@ -66,37 +90,193 @@ public void show() {
...
+    private void init() {
+
+        // set up table headers
+        insertHeaderRow("Action", "Format", "Example", generalPage);
+        insertHeaderRow("Action", "Format", "Example", personsPage);
+        insertHeaderRow("Action", "Format", "Example", appointmentsPage);
+        insertHeaderRow("Action", "Format", "Example", tagsPage);
+
+        // populate general page
+        rowCounter = 1;
```
Comment (KwanHW) [inline src/main/java/seedu/contax/ui/HelpWindow.java:129]: Would it better if these `insertRow()` statements can be abstracted into a general function which fills in each row entry of the pane?
```suggestion
        fillRows(GridPane page, String ...columns)
        // or fillRows(GridPane page, String[] ...table)
```

**Review by KwanHW — COMMENTED** [review]
LGTM. Just made a suggestion to improve the readability and abstraction for filling up the help table.

**Review by dandaandaaaaaan — CHANGES_REQUESTED** [review]
some formatting nits and one issue is on different sized screens boxes with the text too long will cut off as shown in the screenshot. I suggest doing either dynamically sized boxes or just larger boxes in general.
Screenshot 2022-03-24 at 12 29 51 AM

Furthermore, i think we can provide a clickable hyperlink to the user guide page on this screen as well, would be convenient for the user. (not like how they did it with a copy url button, would be better if it was just a clickable link and open with user default browser), but if this is too hard then just forget it its fine

As for UI wise, I think it looks fine. I considered potentially syncing the colours used in #214, but that one will be kind of troublesome and not really too useful so it's just food for thought

File: src/main/java/seedu/contax/ui/HelpWindow.java (hunk 1 above)
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/ui/HelpWindow.java:129]: Thanks for the suggestion! Will consider it

File: src/main/java/seedu/contax/ui/HelpWindow.java [hunk 2]
```diff
@@ -3,29 +3,53 @@
...
 import javafx.fxml.FXML;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
 import javafx.scene.control.Button;
 import javafx.scene.control.Label;
-import javafx.scene.input.Clipboard;
-import javafx.scene.input.ClipboardContent;
+import javafx.scene.control.TextArea;
+import javafx.scene.layout.*;
```
Comment (dandaandaaaaaan) [inline src/main/java/seedu/contax/ui/HelpWindow.java:12]: avoid wildcard imports

File: src/main/java/seedu/contax/ui/HelpWindow.java [hunk 3]
```diff
@@ -66,37 +90,193 @@ public void show() {
         logger.fine("Showing help page about the application.");
         getRoot().show();
         getRoot().centerOnScreen();
+
     }
 
     /**
-     * Returns true if the help window is currently being shown.
+     * Returns true if the help window is currently being shown
```
[the remark this answers is not in the document]
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/ui/HelpWindow.java:97]: Fixed

File: src/main/java/seedu/contax/ui/HelpWindow.java [hunk 4]
```diff
@@ -66,37 +90,193 @@ public void show() {
...
      */
     public boolean isShowing() {
         return getRoot().isShowing();
     }
 
     /**
-     * Hides the help window.
+     * Hides the help window
```
[the remark this answers is not in the document]
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/ui/HelpWindow.java:104]: Fixed

File: src/main/java/seedu/contax/ui/HelpWindow.java [hunk 5]
```diff
@@ -66,37 +90,193 @@ public void show() {
...
-     * Focuses on the help window.
+     * Focuses on the help window
      */
     public void focus() {
         getRoot().requestFocus();
     }
 
     /**
-     * Copies the URL to the user guide to the clipboard.
+     * Initializes the HelpWindow by populating it with data
```
[the remark this answers is not in the document]
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/ui/HelpWindow.java:118]: Fixed

File: src/main/resources/view/HelpWindow.css [hunk 6]
```diff
@@ -1,19 +1,9 @@
...
-}
-
-#helpMessageContainer {
-    -fx-background-color: derive(#1d1d1d, 20%);
+.button {
+    -fx-background-color: #555555;
+    -fx-text-fill: #FFFFFF;
+     -fx-padding: 10px;
+     -fx-border-color: #555555;
+     -fx-border-radius: 5px;
```
[the remark this answers is not in the document]
Comment (Wei-Xinn) [reply src/main/resources/view/HelpWindow.css:6]: Fixed

File: src/main/resources/view/HelpWindow.fxml [hunk 7]
```diff
@@ -1,44 +1,71 @@
...
-<?import javafx.scene.control.Label?>
-<?import javafx.scene.image.Image?>
-<?import javafx.scene.layout.HBox?>
-<?import javafx.stage.Stage?>
+<?import java.net.*?>
+<?import javafx.scene.*?>
+<?import javafx.scene.control.*?>
+<?import javafx.scene.image.*?>
+<?import javafx.scene.layout.*?>
+<?import javafx.stage.*?>
```
[the remark this answers is not in the document]
Comment (Wei-Xinn) [reply src/main/resources/view/HelpWindow.fxml:8]: Fixed

File: src/main/java/seedu/contax/ui/HelpWindow.java (hunk 2 above)
Comment (Wei-Xinn) [reply src/main/java/seedu/contax/ui/HelpWindow.java:12]: FIxed

**Review by dandaandaaaaaan — APPROVED** [review]
lgtm thanks

**Review by sharpstorm — APPROVED** [review]
LGTM.

**Pull request merged**